### PR TITLE
Fix/duplicated tokens

### PR DESCRIPTION
**Pull request by cythb**

### CODE REVIEW

File: Maskbook/Wallet/WalletCore/WalletCoreStorage.swift [hunk 1]
```diff
@@ -166,7 +167,9 @@ extension WalletCoreStorage {
                 try viewContext.saveOrRollback()
                 return false
             }
-            let account = Account(session: session, address: address)
+            // It coverts some abnomal address to standard. The address from metamask is all lowered case.
```
Comment (foxsin10) [inline Maskbook/Wallet/WalletCore/WalletCoreStorage.swift]: Maybe `abnormal` and `The addresses from metamask are all in lowercase`?

File: Maskbook/Supporting Files/AppDelegate.swift [hunk 2]
```diff
@@ -209,6 +212,41 @@ extension AppDelegate {
             }
             .store(in: &disposeBag)
     }
+    
+    // If you connect to the wallet from imToken, metaMask separately, it may result in duplicate tokens, so let's clean up all the tokens first.
+    private func cleanTokens() {
+        guard !userDefaultSettings.hasCleanTokensForBugfix else {
```
Comment (foxsin10) [inline Maskbook/Supporting Files/AppDelegate.swift]: Let's say, the user first import a normal wallet and get tokens, later again import a wallet which is from imToken or MetaMask, will this solution also fix the issue?
Also, we should move this `cleanTokens` function into a repository object.

File: Maskbook/Wallet/WalletCore/WalletCoreStorage.swift (hunk 1 above)
Comment (cythb) [reply Maskbook/Wallet/WalletCore/WalletCoreStorage.swift]: Yes. The abnormal case is that the addresses returned by metamask are all lowercase.

File: Maskbook/Supporting Files/AppDelegate.swift (hunk 2 above)
Comment (cythb) [reply Maskbook/Supporting Files/AppDelegate.swift]: Okay, I'll create a repo in provider directory.

Comment (cythb) [reply Maskbook/Supporting Files/AppDelegate.swift]: We can put the operations of CRUD into the `Repository` and consider putting them into the `Service` if they are more complex. `Service` can implement business logic using the `Repository` methods.

**Review by foxsin10 — APPROVED** [review]

**Pull request merged**